src/main/java/org/opensearch/neuralsearch/processor/util/ProcessorUtils.java

```diff
@@ -171,6 +186,57 @@ public static Optional<Object> getValueFromSource(final Map<String, Object> sour
         return currentValue;
     }
 
+    public static void setValueToSource(Map<String, Object> sourceAsMap, String targetKey, Object targetValue) {
+        setValueToSource(sourceAsMap, targetKey, targetValue, -1);
+    }
+
+    /**
+     * Inserts or updates a value in a nested map structure, with optional support for list traversal.
+     * This method navigates through the provided sourceAsMap using the dot-delimited key path
+     * specified by targetKey. Intermediate maps are created as needed. When a List is encountered,
+     * the provided index is used to select the element from the list. The selected element must be a map to
+     * continue the traversal.
+     * Once the final map in the path is reached, the method sets the value for the last key.
+     *
+     * @param sourceAsMap The Source map (a map of maps) to iterate through
+     * @param targetKey   he path to key to insert the desired targetValue
+     * @param targetValue the value to set at the specified key path
+     * @param index       the index to use when a list is encountered during traversal; if list processing is not needed,
+     *                    -1 is passed in
+     */
+
+    public static void setValueToSource(Map<String, Object> sourceAsMap, String targetKey, Object targetValue, int index) {
+        if (sourceAsMap == null || targetKey == null) return;
+
+        String[] keys = targetKey.split("\\.");
+        Map current = sourceAsMap;
+
+        for (int i = 0; i < keys.length - 1; i++) {
+            Object next = current.computeIfAbsent(keys[i], k -> new HashMap<>());
+            if (next instanceof List<?> list) {
+                if (index < 0 || index >= list.size()) return;
+                current = (Map) list.get(index);
```

Comment:
Please avoid using get when we don't know if list is arraylist. 
Also, casting it to Map is not safe.